bin/brew

```diff
@@ -1,14 +1,27 @@
 #!/bin/bash
 
-chdir() {
-  cd "$@" >/dev/null || {
-    echo "Error: failed to cd to " "$@" "!" >&2
+odie() {
+    if [[ -t 2 ]]
+    then
+        echo -ne "\033[4;31mError\033[0m: " >&2
+    else
+        echo -n "Error: " >&2
+    fi
+    if [[ $# -eq 0 ]]
+    then
+        /bin/cat >&2
+    else
+        echo "$*" >&2
+    fi
     exit 1
-  }
+}
+
+chdir() {
+  cd "$@" >/dev/null || odie "Error: failed to cd to $*!"
 }
 
 # Force UTF-8 to avoid encoding issues for users with broken locale settings.
-if [ "$(locale charmap 2> /dev/null)" != "UTF-8" ]
+if [[ "$(locale charmap 2> /dev/null)" != "UTF-8" ]]
```

Comment:
I see value in only using one such construct instead of having to remember three of them (`[`, `test`, `[[`) and their slightly different semantics and argument interpretation. Given that, to my knowledge, `[[` is the most powerful of them and covers all of the others' features, I'm obviously in support of using it everywhere.